### PR TITLE
Feature/init storybook (#35)

### PR DESCRIPTION
**Pull request by rishiosaur**
* Update package

* Add deps to lockfile

* Set up .storybook pages

* Format App

* Add storybook asset templates

* Recreate package

* update bugfix

* Storybook files

* Update lockfile

* Update imports

* Add new deps to package

* WithTheme HOC

* Base Typography

* Checking App

* Typography Stories

* Delete base stories

* Update deps for Yarn lockfile

* Message Component

* CommunityBar

* Add prettier quoting

* Update app

* Add testing modules

* update actions

* update actions

* stuff

* formatting

* update package.json to fix lint scripts

* lint

* lint fixes

* Update eslint and prettier caches

* Update all deps

* Formatting

* Basic changes

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": [
+
```
Comment (safinsingh) [inline .eslintrc:3]: newline

File: .eslintrc [hunk 2]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": [
+    
     "wesbos",
     "plugin:@typescript-eslint/eslint-recommended",
-    "plugin:@typescript-eslint/recommended"
+    "plugin:@typescript-eslint/recommended",
+
```
Comment (safinsingh) [inline .eslintrc:7]: newline

File: .eslintrc [hunk 3]
```diff
@@ -29,6 +32,7 @@
   },
   "parser": "@typescript-eslint/parser",
   "plugins": [
-    "@typescript-eslint/eslint-plugin"
+    "@typescript-eslint/eslint-plugin",
+
```
Comment (safinsingh) [inline .eslintrc:36]: newline

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -9,4 +9,4 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
       - run: yarn install --frozen-lockfile
-      - run: yarn run lint-workflow
+      - run: yarn run lint:workflow
```
Comment (Muirrum) [inline .github/workflows/lint.yml:12]: Is this part of the storybook?

File: .github/workflows/format.yml [hunk 5]
```diff
@@ -9,4 +9,4 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
       - run: yarn install --frozen-lockfile
-      - run: yarn run format-check
+      - run: yarn run format:check
```
Comment (safinsingh) [inline .github/workflows/format.yml:12]: if u want u can change check to ci (it sounds cooler lol)

File: .github/PULL_REQUEST_TEMPLATE/BUGFIX.md [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ---
-name: '🐛⚡️ New Fix: <component-slug>'
+name: '🐛⚡️ New Fix: <bug-slug>'
 about: New bugfix for the Banur web interface
```
Comment (Muirrum) [inline .github/PULL_REQUEST_TEMPLATE/BUGFIX.md:3]: Is this part of storybook?

File: .prettierrc [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
-	"arrowParens": "always",
-	"endOfLine": "lf",
-	"jsxBracketSameLine": true,
-	"useTabs": true,
-	"singleQuote": true
+  "arrowParens": "always",
```
Comment (safinsingh) [inline .prettierrc:2]: these look 2 spaces (on github, can u confirm that they're tabs?

**Review by Muirrum — CHANGES_REQUESTED** [review]

File: .storybook/preview.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (safinsingh) [inline .storybook/preview.js:1]: newline

File: .github/workflows/lint.yml (hunk 4 above)
Comment (rishiosaur) [reply .github/workflows/lint.yml:12]: I just changed some of the scripts to keep the style—not strictly a part of storybook but I didn't think it was that big of a fix that necessitated a whole new branch

File: src/components/molecules/Message/Message.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Text } from '@geist-ui/react';
+
+console.log('hi');
+export const Title: React.FC = ({ children, ...props }) => (
+	<Text h1 {...props}>
```
Comment (safinsingh) [inline src/components/molecules/Message/Message.stories.tsx:6]: make sure prop spreading is whitelisted in eslint

**Review by safinsingh — CHANGES_REQUESTED** [review]
couple small things, btw, when i say newline i meant "extra newline, pls remove"

File: .prettierrc [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
+  "endOfLine": "lf",
+  "jsxBracketSameLine": true,
+  "useTabs": true,
+  "singleQuote": true
 }
```
Comment (safinsingh) [inline .prettierrc:7]: oh and can we please have `semi: false`

**Review by safinsingh — CHANGES_REQUESTED** [review]

**Review by Muirrum — CHANGES_REQUESTED** [review]
In general there's a lot of stuff that should be separated into another pr (like the linting and CI stuff)

**Comment by rishiosaur** [comment]
Yeah it really should lol

**Comment by rishiosaur** [comment]
I'm just setting things up rn tho; after this PR everything should be clean

**Comment by safinsingh** [comment]
will approve as soon as ci passes!

**Review by safinsingh — APPROVED** [review]
yeet!

**Review by safinsingh — APPROVED** [review]

**Pull request merged**